code/ACLoader.cpp

```diff
@@ -584,9 +587,19 @@ aiNode* AC3DImporter::ConvertObjectSection(Object& object,
 
                 // allocate storage for vertices and normals
                 mesh->mNumFaces = (*cit).first;
+                if (mesh->mNumFaces == 0) {
+                    throw DeadlyImportError("AC3D: No faces");
+                } else if (mesh->mNumFaces > std::numeric_limits<int32_t>::max() / sizeof(aiFace)) {
```

Comment:
This check looks like it presume the boundary of the 32 bit address space? We would run out of memory (both physical as well as virtual space) much earlier.

I'd suggest setting a constant upper bound of i.e. 100 million.